ios/brave-ios/Sources/Brave/Frontend/Browser/Translate/TranslateToast.swift

```diff
@@ -0,0 +1,361 @@
+// Copyright 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import BraveUI
+import DesignSystem
+import SwiftUI
+
+private struct TranslationOptionsView: View {
+  @Environment(\.dismiss)
+  private var dismiss
+
+  @State
+  private var searchText = ""
+
+  @State
+  private var timer: Timer?
+
+  @Binding
+  var language: Locale.Language?
+
+  var body: some View {
+    NavigationStack {
+      List {
+        ForEach(filteredLanguages, id: \.self) { language in
+          Button(
+            action: {
+              self.language = language
+              dismiss()
+            },
+            label: {
+              Text(languageName(for: language))
+            }
+          )
+        }
+      }
+      .listStyle(.plain)
+      .navigationTitle("Select Language")
+      .navigationBarTitleDisplayMode(.inline)
+      .toolbar {
+        ToolbarItem(placement: .navigationBarLeading) {
+          Button("Cancel") {
+            dismiss()
+          }
+        }
+      }
+    }
+    .searchable(
+      text: $searchText,
+      placement: .navigationBarDrawer(displayMode: .always),
+      prompt: "Search"
+    )
+    .onChange(of: searchText) { searchText in
+      self.timer?.invalidate()
+      self.timer = Timer.scheduledTimer(
+        withTimeInterval: 0.1,
+        repeats: false,
+        block: { timer in
+          timer.invalidate()
+          // TODO: Filter here
+        }
+      )
+    }
+  }
+
+  private var filteredLanguages: [Locale.Language] {
+    return searchText.isEmpty
+      ? languages
+      : languages.filter({
+        languageName(for: $0).lowercased().contains(searchText.lowercased())
+      })
+  }
+
+  // TODO: Take from Brave-Core's list
+  // TODO: Take from Apple's list
+  private var languages: [Locale.Language] {
+    return [
```

Comment:
as the TODO mentions do we need to hardcode the language list?

Reply:
Yes. For now. Once we move to Brave-Core we'll be getting rid of this anyway and it'll come from Brave-Core. Otherwise I have to export the list.